scripts/core/editor3/directive.tsx

```diff
@@ -296,22 +327,171 @@ class Editor3Directive {
+                                            justifyContent: 'space-between',
+                                            alignItems: 'center',
+                                        }}
+                                    >
+                                        <div
+                                            style={{
+                                                display: 'flex',
+                                                gap: 8,
+                                                alignItems: 'center',
+                                            }}
+                                        >
+                                            <div className="field__label" style={labelStyle}>{fieldName}</div>
+                                            {this.required && (
+                                                <span className="sd-required">{gettext('Required')}</span>
+                                            )}
+                                        </div>
+
+                                        {textStatistics}
+                                    </div>
+
+                                    {validationErrors}
+                                </div>
+                                {editor3}
+
+                                <div className="sd-editor__info-text">{this.helperText}</div>
+                            </div>
+                        );
+                    };
+
+                    const getTemplateForHeader = () => {
+                        return (
+                            <div style={{display: 'flex'}}>
+                                <div className="authoring-header__item-label">
+                                    {fieldName}
+                                    {this.required && (
+                                        <span>
+                                            &nbsp;
+                                            <span
+                                                aria-label={gettext('required')}
+                                                style={{color: 'red', fontSize: 12}}
+                                            >
+                                                *
+                                            </span>
+                                        </span>
+                                    )}
+                                </div>
+
+                                <div style={{flexGrow: 1}}>
+                                    <div>
+                                        {editor3}
+                                    </div>
+
+                                    <div
+                                        style={{
+                                            display: 'flex',
+                                            justifyContent: 'space-between',
+                                            alignItems: 'center',
+                                        }}
+                                    >
```

Comment:
Use a Spacer, if you can't directly it might be a good idea for us to bridge it, so it can be used in Angular too